### PR TITLE
[MXNET-908] Enable minimal OSX Travis build (v1.3.x)

### PR DESCRIPTION
**Pull request by lebeg**
## Description ##

Port of https://github.com/apache/incubator-mxnet/pull/12462 to the v1.3.x release branch.

Includes following PR's as well:

[MXNET-908] Enable python tests in Travis
https://github.com/apache/incubator-mxnet/pull/12550

[MXNET-968] Fix MacOS python tests
https://github.com/apache/incubator-mxnet/pull/12590

And due to timeout:

Disable travis tests
https://github.com/apache/incubator-mxnet/pull/13137

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by lebeg** [comment]
@KellenSunderland @nswamy @szha

**Comment by marcoabreu** [comment]
There are a few more Travis related PRs. Please add these commits as well as they ensure high stability of Travis.

File: .travis.yml [hunk 1]
```diff
@@ -1,108 +1,32 @@
...
-  - tests/travis/travis_after_failure.sh
-
-notifications:
-# Emails are sent to the committer's git-configured email address by default,
-  email:
-    on_success: change
-    on_failure: always
-  #slack: dmlc:NmroCzntCiWOuxUZpii40USd
+  - export MXNET_STORAGE_FALLBACK_LOG_VERBOSE=0
+  - mv make/osx.mk config.mk
```
Comment (larroy) [inline .travis.yml]: CMake?

Comment (lebeg) [reply .travis.yml]: I don't want to modify code for now, just porting things.

**Comment by nswamy** [comment]
I agree with @marcoabreu lets not add it to the branch

**Comment by lebeg** [comment]
@marcoabreu @nswamy I think the timeout is not happening on the branch. Let's merge and observe, this will not harm, but in case it will work has the benefit to cover MacOS with testing.

File: src/initialize.cc [hunk 2]
```diff
@@ -60,7 +60,7 @@ class LibraryInitializer {
         // Make children single threaded since they are typically workers
         dmlc::SetEnv("MXNET_CPU_WORKER_NTHREADS", 1);
         dmlc::SetEnv("OMP_NUM_THREADS", 1);
-#if MXNET_USE_OPENCV
+#if MXNET_USE_OPENCV && !__APPLE__
```
Comment (szha) [inline src/initialize.cc]: What's the reason for not disabling opencv threads on mac?

**Comment by nswamy** [comment]
@lebeg I am not seeing a need for the OS X builds since the tests are disabled and also this did not exist for 1.3.0

**Comment by lebeg** [comment]
@nswamy Ok, for clarity I understand this as: *Strongly opposed to have it in the release branch*. I'm not seeing that much risk as you do, but I commit to this decision and close this PR.

**Pull request closed without merging**